docs/src/index.md

```diff
@@ -14,70 +14,46 @@ Whenever one of your package's dependencies releases a new breaking version, Com
-
-The default CompatHelper setup has one flaw: the pull requests that CompatHelper opens will not trigger any other GitHub Actions.
-
-Consider the following situations:
-1. You use GitHub Actions to test your package using continuous integration (CI).
-2. You use GitHub Actions to build and deploy the documentation for your package.
-
-If any of those situations apply to you, then you will need to set up an SSH deploy key for CompatHelper. Once you have set up an SSH deploy key for CompatHelper, the pull requests that CompatHelper opens will trigger all of the usual GitHub Actions.
-
-If none of those situations apply to you, then you don't need to set up an SSH deploy key for CompatHelper.
-
-### Instructions for setting up the SSH deploy key
-
-It is easy to set up an SSH deploy key for CompatHelper. Here are the instructions:
-1. `ssh-keygen -m PEM -N "" -f compathelper_key`
-2. `cat compathelper_key # This is the private key in PEM format. Copy this to your clipboard.`
-3. Go to the GitHub page for your package's repository, click on the **Settings** tab, then click on **Secrets**, and then click on **New repository secret**. Name the secret `COMPATHELPER_PRIV`. For the contents, paste in the PEM-formatted private key that you copied in the previous step. Make sure that you include the entire PEM-formatted private key, including the lines at the beginning and end that start with `-----BEGIN` and `-----END` respectively.
-4. `cat compathelper_key.pub # This is the public key. Copy this to your clipboard.`
-5. Go to the GitHub page for your package's repository, click on the **Settings** tab, then click on **Deploy keys**, and then click on **Add deploy key**. Name the deploy key `COMPATHELPER_PUB`. For the contents, paste in the public key that you copied in the previous step. Make sure that you give the key **write access**.
-6. `rm -f compathelper_key compathelper_key.pub`
-
-### Advanced notes
-
-When you supply the private key, you can either provide the raw PEM-formatted private key itself (as we did above), or you can provide the Base64-encoded form of the PEM-formatted private key.
-
-For example, if you already have a Base64-encoded PEM-formatted private key saved as a secret, you can re-use it. If e.g. the secret is named `DOCUMENTER_KEY`, then simply replace the line that looks like this:
-```yaml
-COMPATHELPER_PRIV: ${{ secrets.COMPATHELPER_PRIV }}
-```
-
-with this line:
-```yaml
-COMPATHELPER_PRIV: ${{ secrets.DOCUMENTER_KEY }}
+name: CompatHelper
+on:
+  schedule:
+    - cron: '0 0 * * *'  # Everyday at midnight
+  workflow_dispatch:
+jobs:
+  CompatHelper:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Pkg.add("CompatHelper")
+        run: julia -e 'using Pkg; Pkg.add("CompatHelper")'
+      - name: CompatHelper.main()
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COMPATHELPER_PRIV: ${{ secrets.COMPATHELPER_PRIV }}
+        run: julia -e 'using CompatHelper; CompatHelper.main()'
```

Comment:
I'm not sure if it makes more sense to use this PkgTemplates version or to use the version that exists in this repo. The one in this repo is possibly better because it pins it to the major version, but on the other hand people probably don't want to have to update all their packages whenever a major change comes out. @DilumAluthge any comments on this?

Reply:
I had taken this from what already existed, but can def change it up!

Reply:
yeah the one you have there is the one that exists in PkgTemplates, but the one that exists for this repo is here https://github.com/JuliaRegistries/CompatHelper.jl/blob/master/.github/workflows/CompatHelper.yml

Reply:
So, a couple of notes:

1. Regarding PkgTemplates, we (the CompatHelper.jl repo) are the upstream here, and thus we are the authoritative source. So the workflow file that exists in the CompatHelper.jl repo at `.github/workflows/CompatHelper.yml` is the authoritative workflow file. If the PkgTemplates version differs from our version, we should probably make a PR to edit the PkgTemplate version so that it exactly matches ours.
2. Yeah, it is intentional that our workflow file (`.github/workflows/CompatHelper.yml`) pins the major version. This means that, in the future, we (CompatHelper maintainers) can make breaking changes if we really want, and it won't break the whole ecosystem. It's a little inconvenient that people will have to change e.g. "2" to "3", but the alternative is that if we make a breaking change, hundreds of people start getting emails that their workflows are broken, which leads to a lot of irate users. In fact I did this once, and people were not happy 😂 .
3. Regarding the documentation, I'd like us to avoid repeating ourselves. That is, I don't want to have the same content in multiple places, because that means we have to update multiple locations. So the `.github/workflows/CompatHelper.yml` file should be the only place that content is defined. If we want to use that content elsewhere (e.g. in the docs), we should just include it from the `.github/workflows/CompatHelper.yml`. If you look how we do it in the docs on master:

https://github.com/JuliaRegistries/CompatHelper.jl/blob/0afb9df0164b9389ffbd444978b7e9c2e61fa5b3/docs/src/index.md#L15-L28

Reply:
That makes sense to me. If it exists in the project we should just link it. I agree on using the version of CompatHelper.yml that we have in this project with the version specifically named. We should make an issue on PkgTemplates to get that changed.